src/common/src/array/primitive_array.rs

```diff
@@ -132,10 +132,22 @@ pub struct PrimitiveArray<T: PrimitiveArrayItemType> {
     data: Vec<T>,
 }
 
-impl<T: PrimitiveArrayItemType> PrimitiveArray<T> {
-    pub fn from_slice(data: &[Option<T>]) -> Self {
-        let mut builder = <Self as Array>::Builder::new(data.len());
-        for i in data {
+impl<T: PrimitiveArrayItemType> FromIterator<T> for PrimitiveArray<T> {
+    fn from_iter<I: IntoIterator<Item = T>>(iter: I) -> Self {
```

Comment:
How about forwarding the implementation to `FromIterator<&'a Option<T>>`? 🤔

Reply:
I tried but failed due to lifetime problem. 🥹
![WX20221209-131015@2x](https://user-images.githubusercontent.com/15158738/206628636-4d7366d5-49bb-475b-bd4e-bf572a0264ad.png)

Reply:
But we could forward both `FromIterator<T>` and `FromIterator<&'a Option<T>>` to `FromIterator<Option<T>>`.